### PR TITLE
THREESCALE-7195 remove wip to check faliure

### PR DESCRIPTION
**Pull request by nidhi-soni1104**
**This PR Fixes @wip feature for BulkOperations for this file , which are failing in circle CI**
porta/features/old/accounts/bulk_operations.feature
https://issues.redhat.com/browse/THREESCALE-9757

### CODE REVIEW

File: features/old/accounts/bulk_operations.feature [hunk 1]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
     Given current domain is the admin domain of provider "foo.3scale.localhost"
     Given I am logged in as provider "foo.3scale.localhost"
 
-  # FIXME: THREESCALE-7195 this scenario is failing in CircleCI. We need to refactor it as an integration test.
-  @wip
   Scenario: Show and hide bulk operations controls
     When I navigate to the accounts page
     When I check select for "bob"
-    Then "Bulk operations" should be visible
+    And checks if bulk-operations the div is visible
```
Comment (jlledom) [inline features/old/accounts/bulk_operations.feature:23]: I think cucumbers describe which should be the user experience, for that reason I think this step shouldn't include any reference to the `div` in it's name. I think cucumber already provide an `is visible` step. You can also use `I should see`/`I should not see`.

File: features/old/accounts/bulk_operations.feature [hunk 2]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
       And I should see "Send email"
       And I should see "Change account plan"
       And I should see "Change state"
     When I uncheck select for "bob"
-    Then "Bulk operations" should not be visible
+    And checks if bulk-operations the div is hidden
```
Comment (jlledom) [inline features/old/accounts/bulk_operations.feature:28]: Same here.

File: features/old/accounts/bulk_operations.feature [hunk 3]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
-  # FIXME: THREESCALE-7195 this scenario is failing in CircleCI. We need to refactor it as an integration test.
-  @wip
   Scenario: Check all accounts with main checkbox
       And I am on the accounts admin page
 
     When I check select in table header
-
-    Then all selects should be checked
-      And "Bulk operations" should be visible
+    Then I check if all checkboxes are selected
```
Comment (jlledom) [inline features/old/accounts/bulk_operations.feature:34]: This step doesn't exist, so the old name `all selects should be checked` is available. Why did you rename it? I think the old one is better.

File: features/old/accounts/bulk_operations.feature [hunk 4]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
+      And checks if bulk-operations the div is visible
 
     When I uncheck select in table header
-    Then none selects should be checked
-      And "Bulk operations" should not be visible
+    Then I check if none of the checkboxes are selected
```
Comment (jlledom) [inline features/old/accounts/bulk_operations.feature:38]: This could be called `No selects should be checked`

**Comment by nidhi-soni1104** [comment]
@jlledom if you see these are wip test, means they are incomplete and for few steps definitons are not defined .

i have defined and updated the test

File: features/old/accounts/bulk_operations.feature (hunk 3 above)
Comment (nidhi-soni1104) [reply features/old/accounts/bulk_operations.feature:34]: ok

File: features/old/accounts/bulk_operations.feature (hunk 4 above)
Comment (nidhi-soni1104) [reply features/old/accounts/bulk_operations.feature:38]: ok I will update this

File: features/old/accounts/bulk_operations.feature (hunk 1 above)
Comment (jlledom) [reply features/old/accounts/bulk_operations.feature:23]: Yes, it can check for a div internally, but the name shouldn't mention the div IMO, it should be something more closer to the user experience like `I should see the Bulk opertaions panel`

Comment (mayorova) [reply features/old/accounts/bulk_operations.feature:23]: I agree with what @jlledom says. And it's true that just rewording these two lines as `Then I should see "Bulk operations"` and `Then I should not see "Bulk operations"` should be enough, I think, without adding the new steps.

Also, the `When` online 22 can be changed to `And` for sake for nicer wording :)

File: features/old/accounts/bulk_operations.feature [hunk 5]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
       And I should see "Change account plan"
       And I should see "Change state"
     When I uncheck select for "bob"
-    Then "Bulk operations" should not be visible
+    And checks if bulk-operations the div is hidden
 
-  # FIXME: THREESCALE-7195 this scenario is failing in CircleCI. We need to refactor it as an integration test.
-  @wip
   Scenario: Check all accounts with main checkbox
       And I am on the accounts admin page
```
Comment (mayorova) [inline features/old/accounts/bulk_operations.feature:31]: I'd improve the steps readability by following the order:
```
When ...
And ...
Then ...
```

File: features/step_definitions/applications/bulk_operations_steps.rb [hunk 6]
```diff
@@ -51,3 +51,21 @@ def check_or_uncheck_selected_row value, text = nil
...
+  assert_selector('#bulk-operations', visible: :hidden)
+end
+
+And(/^checks if bulk-operations the div is visible$/) do
+  assert_selector('#bulk-operations', visible: true)
+end
+
+Then(/^I check if none of the checkboxes are selected$/) do
+  checkboxes = all('input[type="checkbox"]')
+  expect(checkboxes.all? { |checkbox| !checkbox.checked? }).to be true
```
Comment (mayorova) [inline features/step_definitions/applications/bulk_operations_steps.rb:65]:
```suggestion
  assert checkboxes.none?(&:checked?)
```

File: features/step_definitions/applications/bulk_operations_steps.rb [hunk 7]
```diff
@@ -51,3 +51,21 @@ def check_or_uncheck_selected_row value, text = nil
...
+end
+
+Then(/^I check if all checkboxes are selected$/) do
+  checkboxes = all('input[type="checkbox"]')
+  expect(checkboxes.all?(&:checked?)).to be true
```
Comment (mayorova) [inline features/step_definitions/applications/bulk_operations_steps.rb:70]:
```suggestion
  assert checkboxes.all?(&:checked?)
```

**Comment by nidhi-soni1104** [comment]
@jlledom @mayorova all comments are addressed

File: features/old/accounts/bulk_operations.feature [hunk 8]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
   Scenario: Check all accounts with main checkbox
-      And I am on the accounts admin page
-
-    When I check select in table header
+    When I am on the accounts admin page
 
+    And I check select in table header
     Then all selects should be checked
-      And "Bulk operations" should be visible
+    Then I should see "Bulk operations"
```
Comment (mayorova) [inline features/old/accounts/bulk_operations.feature:35]:
```suggestion
    And I should see "Bulk operations"
```

File: features/old/accounts/bulk_operations.feature [hunk 9]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
 
-    When I uncheck select in table header
-    Then none selects should be checked
-      And "Bulk operations" should not be visible
+    And I uncheck select in table header
```
Comment (mayorova) [inline features/old/accounts/bulk_operations.feature:37]:
```suggestion
    When I uncheck select in table header
```

File: features/old/accounts/bulk_operations.feature [hunk 10]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
+    Then No selects should be checked
```
Comment (mayorova) [inline features/old/accounts/bulk_operations.feature:38]:
```suggestion
    Then no selects should be checked
```

File: features/old/accounts/bulk_operations.feature [hunk 11]
```diff
@@ -17,28 +17,23 @@ Feature: Bulk operations
...
+    Then I should not see "Bulk operations"
```
Comment (mayorova) [inline features/old/accounts/bulk_operations.feature:39]:
```suggestion
    And I should not see "Bulk operations"
```

File: features/old/services/subscriptions/bulk_operations.feature [hunk 12]
```diff
@@ -21,28 +21,25 @@ Feature: Bulk operations
...
     When I check select in table header
 
     Then all selects should be checked
-      And "Bulk operations" should be visible
+    Then I should see "Bulk operations"
 
     When I uncheck select in table header
-    Then none selects should be checked
-      And "Bulk operations" should not be visible
+    Then No selects should be checked
```
Comment (mayorova) [inline features/old/services/subscriptions/bulk_operations.feature:44]:
```suggestion
    Then no selects should be checked
```

File: features/old/services/subscriptions/bulk_operations.feature [hunk 13]
```diff
@@ -21,28 +21,25 @@ Feature: Bulk operations
...
+    Then I should not see "Bulk operations"
```
Comment (mayorova) [inline features/old/services/subscriptions/bulk_operations.feature:45]:
```suggestion
    And I should not see "Bulk operations"
```

File: features/step_definitions/applications/bulk_operations_steps.rb [hunk 14]
```diff
@@ -51,3 +51,13 @@ def check_or_uncheck_selected_row value, text = nil
 When "I should see the bulk action failed with {application}" do |application|
   assert_match "There were some errors:\n#{application.name} (#{application.user_account.org_name})", bulk_errors_container.text
 end
+
+Then(/^No selects should be checked$/) do
```
Comment (mayorova) [inline features/step_definitions/applications/bulk_operations_steps.rb:55]:
```suggestion
Then(/^no selects should be checked$/) do
```

File: features/step_definitions/applications/bulk_operations_steps.rb [hunk 15]
```diff
@@ -51,3 +51,13 @@ def check_or_uncheck_selected_row value, text = nil
...
+  checkboxes = all('input[type="checkbox"]')
+  assert checkboxes.none?(&:checked?)
+end
+
+Then(/^all selects should be checked$/) do
+  checkboxes = all('input[type="checkbox"]')
+   assert checkboxes.all?(&:checked?)
```
Comment (mayorova) [inline features/step_definitions/applications/bulk_operations_steps.rb:62]:
```suggestion
  assert checkboxes.all?(&:checked?)
```

**Comment by nidhi-soni1104** [comment]
@mayorova Thanks for clearing the order of when then . I committed them all

**Review by jlledom — APPROVED** [review]
